overlord/devicestate/devicemgr.go

```diff
@@ -991,6 +992,55 @@ func (m *DeviceManager) ensureSeeded() error {
 	return nil
 }
 
+var processAutoImportAssertionsImp = processAutoImportAssertions
+
+// ensureAutoImportAssertions makes sure that auto import assertions
+// get processed
+func (m *DeviceManager) ensureAutoImportAssertions() {
+	// on classic do nothing
+	if release.OnClassic {
+		return
+	}
+
+	// ignore while we are in preseeding
+	if m.preseed {
+		return
+	}
+
+	// check if we have processed auto import as already
+	var asState string
+	err := m.state.Get("auto-as-imported", &asState)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		logger.Noticef("failed to get auto-import assert state")
+		return
+	}
+
+	if asState == "done" || asState == "pending" {
+		return
+	}
+
+	cached := m.state.Cached(loadedDeviceSeedKey{})
+	if cached != nil {
+		loaded := cached.(*loadedDeviceSeed)
+		deviceSeed := loaded.seed
+		commitTo := func(batch *asserts.Batch) error {
+			return assertstate.AddBatch(m.state, batch, nil)
+		}
+		db := assertstate.DB(m.state)
+		err = processAutoImportAssertionsImp(m.state, deviceSeed, db, commitTo)
+		// check error, if add user failed, with missing serial assertion
+		// we mark state as "pending"
+		if err != nil {
+			logger.Noticef("cannot process auto import assertion: %v", err)
+			if err == ErrDeviceNotRegistered {
+				m.state.Set("auto-as-imported", "pending")
+			}
+		} else {
+			m.state.Set("auto-as-imported", "done")
```

Comment:
we should do this also if the error is not nil but is not ErrDeviceRegistered, no?

Reply:
Agree.
I did not set it on purpose, as I wanted to check first with you if there is some other error case when we want to retry.
I will mark it here as `"done"` and if some other cases come in the future when want to retry, we special case as they come.